qaframework-bdd-tests/src/test/resources/features/refapp-2.x/08-encounter/patientEncounter.feature

```diff
@@ -0,0 +1,18 @@
+Feature: Patient Encounter Management
+
+  @selenium
+  @patientEncounter
+  Scenario: Add and Delete Encounter
+    #User story: Add Encounter
+    Given a user clicks on System Administration app from the home page
+    When a user clicks on Advanced Administration app from the System Administration page
+    And a user clicks on Manage Encounters link from the Administration page
```

Comment:
We can have the first three steps put under the background.

Reply:
> We can have the first three steps put under the background.

@kdaud please why would you want us to have the first steps in the background? I see they  are only required once by the Add encounter user story. Advise me on this. The Delete Encounter user story does not require to run those steps.

Reply:
As suggested earlier, we can have the user story of adding an encounter begin with;

```js
And a user clicks on Add Encounter link from the encounters page
    When a user fills the ecounter form
    And a user clicks Save Encounter button
    Then the system saves encounter in the encounters table
```
so that the other first three steps are considered to be the background of the test

Reply:
> The Delete Encounter user story does not require to run those steps

Deleting an encounter can only happen when there is an added encounter in the table.

Reply:
> Deleting an encounter can only happen when there is an added encounter in the table.

@kdaud very true. However, I implemented it in a way that when an encounter is saved, the test does not have to get back to the home page and then look for the encounter. Instead since after saving the encounter, the form does not get dismissed, right there we can check the delete checkbox and implement deletion of the encounter without leaving that page.

Reply:
Does putting the first three steps in the background fails the test?

Reply:
@Irenyak1 it's a good practice to ask for clarity on any review comment which is unclear to you :heart:

Reply:
yeah